Testscripts/Linux/perf_lkp.sh

```diff
@@ -0,0 +1,186 @@
+#!/bin/bash
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache License.
+
+###############################################################################
+#
+# Description:
+# We use LKP framework to run a few benchmarks which includes hackbench, unixbench,
+# perf-bench-numa-mem, perf-bench-sched-pipe, aim9, pmbench.
+#
+# Support distros: Ubuntu, Centos 7, RedHat 7, Debian
+#
+###############################################################################
+
+LKP_SCHEDULER_HACKBENCH_TESTS=("hackbench-100.yaml" "hackbench.yaml")
+LKP_SCHEDULER_PERF_BENCH_SCHED_PIPE_TESTS=("perf-bench-sched-pipe.yaml")
+LKP_MEMORY_PERF_BENCH_NUMA_MEM_TESTS=("perf-bench-numa-mem.yaml")
+LKP_MEMORY_PMBENCH_TESTS=("pmbench.yaml")
+LKP_TESTS_ALL_ABOVE=("perf-bench-sched-pipe.yaml" "perf-bench-numa-mem.yaml" "pmbench.yaml")
+LKP_OTHER_UNIXBENCH_TESTS=("unixbench.yaml")
+LKP_OTHER_AIM9_TESTS=("aim9.yaml")
+
+declare -A TEST_DICT
+TEST_DICT=(["hackbench"]=${LKP_SCHEDULER_HACKBENCH_TESTS[@]} \
+           ["perf-bench-sched-pipe"]=${LKP_SCHEDULER_PERF_BENCH_SCHED_PIPE_TESTS[@]} \
+           ["perf-bench-numa-mem"]=${LKP_MEMORY_PERF_BENCH_NUMA_MEM_TESTS[@]} \
+           ["pmbench"]=${LKP_MEMORY_PMBENCH_TESTS[@]} \
+           ["unixbench"]=${LKP_OTHER_UNIXBENCH_TESTS[@]} \
+           ["aim9"]=${LKP_OTHER_AIM9_TESTS[@]} \
+           ["lkp_microbenchmark"]=${LKP_TESTS_ALL_ABOVE[@]})
+
+###############################################################################
+function install_dependencies() {
+    LogMsg "Installing dependency packages"
+
+    update_repos
+
+    case $DISTRO in
+        ubuntu* | debian*)
+            CheckInstallLockUbuntu
+            deb_packages=(make git)
+            LogMsg "Dependency package names： ${deb_packages[@]}"
+            install_package "${deb_packages[@]}" >> $LKP_OUTPUT 2>&1
+            ;;
+        centos_7 | redhat_7)
+            rpm_packages=(make git curl psmisc)
+            install_epel
+            LogMsg "Dependency package names： ${rpm_packages[@]}"
+            install_package "${rpm_packages[@]}" >> $LKP_OUTPUT 2>&1
+            gpg --keyserver hkp://keys.gnupg.net --recv-keys 409B6B1796C275462A1703113804BB82D39DC0E3 7D2BAF1CF37B13E2069D6956105BD0E739499BDB
+            curl -sSL https://get.rvm.io | bash -s stable
+            if [ $? -ne 0 ]; then
+                LogErr "Failed to curl rvm"
+                SetTestStateAborted
+                exit 1
+            fi
+            source /etc/profile.d/rvm.sh
+            rvm requirements
+            rvm install ruby-2.3.0
+            rvm use ruby-2.3.0 default
+            PATH="$PATH:/usr/local/bin"
+            ;;
+        *)
+            LogErr "Unsupported distro: $DISTRO"
+            return 1
+            ;;
+    esac
+}
+
+function run_tests {
+    LogMsg "git clone LKP..."
+    if [[ $DISTRO =~ "redhat" ]]; then
+        work_dir=$(df -h | grep "/dev/sdb1" | awk -F " " '{ print $6 }')
```

Comment:
it is because that the disk doesn't have enough space? please add some comments here to clarify why need extra steps on redhat, also use the hard-code sdb is not recommended.